src/cortex_m.c

```diff
@@ -1,250 +1,124 @@
 #include <stddef.h>
 #include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 #include "zenroom.h"
+
 #ifdef ARCH_CORTEX
-extern unsigned int _start_heap;
-// #define NULL (((void *)0))
-//
-//
 
+#include <errno.h>
+#undef errno
+extern int errno;
+
+enum semihosting_sys
```

Comment:
Rename to `semihost_ops`.
See https://github.com/ContextualElectronics/stm32-bootloader/blob/master/bootloader/system/include/arm/semihosting.h